LeaderboardBackend/Controllers/LeaderboardsController.cs

```diff
@@ -37,24 +38,27 @@ public async Task<ActionResult<Leaderboard>> GetLeaderboard(long id)
 			return NotFound();
 		}
 
-		return Ok(leaderboard);
+		return Ok(LeaderboardViewModel.MapFrom(leaderboard));
 	}
 
 	/// <summary>
 	///     Gets Leaderboards by their IDs.
 	/// </summary>
 	/// <param name="ids">The IDs of the `Leaderboard`s which should be retrieved.</param>
+	/// <param name="slug">The slug of the `Leaderboard` to retrieve</param>
 	/// <response code="200">
 	///     The list of `Leaderboard`s was retrieved successfully. The result can be an empty
 	///     collection.
 	/// </response>
 	[AllowAnonymous]
 	[HttpGet]
 	[ProducesResponseType(StatusCodes.Status200OK)]
-	public async Task<ActionResult<List<Leaderboard>>> GetLeaderboards(
-		[FromQuery] long[] ids)
+	public async Task<ActionResult<List<LeaderboardViewModel>>> GetLeaderboards(
+		[FromQuery] long[] ids, [FromQuery] string? slug = null)
```

Comment:
There can't ever be two leaderboards with the same slug, right? We could instead have a `GetLeaderboardBySlug` method maybe which returns a `LeaderboardViewModel` instead of a List of them. I'd imagine it'd be annoying for users to find a leaderboard by its slug to then have to get the only element out of the list every time.

I believe we can typehint the parameter in the REST method attribute, which should allow two methods with the "same" `[HttpGet("{id}")]`

(And if this works of course the tests will need to be changed, sorry)

Reply:
Continuation on the issue: https://github.com/leaderboardsgg/leaderboard-backend/issues/136#issuecomment-1528681722